### PR TITLE
Three improvements for the glossary

### PR DESCRIPTION
**Pull request by norosa**
three improvements for the glossary:

- Add more terms to the glossary
- Improve the documentation of binary and unary operators
- Improve cross-references for "value expression" term

see commit messages for details

this is based off of `nomi/gloss-20210302-02`, but once that branch is merged, I will rebase onto `master`

### CODE REVIEW

File: docs/general/dql/index.rst [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+.. _dql:
 
 ========
 Querying
 ========
 
-This section provides an overview on how to query documents using SQL.  See
-:ref:`sql_ddl` for information about `Table creation` and other Data Definition
+This section provides an overview on how to query documents using SQL. See
+:ref:`ddl` for information about `Table creation` and other Data Definition
```
Comment (amotl) [inline docs/general/dql/index.rst:10]: Did you intend to link `Table creation` anywhere?

File: docs/sql/general/value-expressions.rst [hunk 2]
```diff
@@ -80,16 +79,17 @@ Parameter references can either be unnumbered or numbered:
 Operator invocation
 ===================
 
-An :ref:`operator <gloss-operator>` can be invoked as a value expression in two
-ways:
+An :ref:`operator <gloss-operator>` can be invoked as a value expression in one
+of two way: *binary* or *unary*.
```
Comment (amotl) [inline docs/sql/general/value-expressions.rst:83]:
```suggestion
of two ways: *binary* or *unary*.
```

File: docs/sql/general/value-expressions.rst [hunk 3]
```diff
@@ -80,16 +79,17 @@ Parameter references can either be unnumbered or numbered:
...
 
-- Binary: ``expression operator expression``
+The syntax of a binary operator::
 
-- Unary: ``operator expression``
+    expression operator expression
 
-.. SEEALSO::
+The syntax of a unary operator::
```
Comment (amotl) [inline docs/sql/general/value-expressions.rst:89]:
```suggestion
The syntax of an unary operator::
```

**Review by amotl — APPROVED** [review]
👍

File: docs/general/dql/index.rst (hunk 1 above)
Comment (norosa) [reply docs/general/dql/index.rst:10]: that was there before I made this change. however, I took the opportunity to improve the index pages for DDL, DML, and DQL. thanks!

File: docs/sql/general/value-expressions.rst (hunk 3 above)
Comment (norosa) [reply docs/sql/general/value-expressions.rst:89]: this isn't good grammar. "a" is correct

**Comment by norosa** [comment]
@amotl pushed a fixup. thank you!!

File: docs/sql/general/value-expressions.rst (hunk 3 above)
Comment (amotl) [reply docs/sql/general/value-expressions.rst:89]: All right. Thanks!

**Pull request merged**